operators/src/pro/meta/statistics.rs

```diff
@@ -138,6 +139,12 @@ where
         query: QueryRectangle<Self::SpatialBounds>,
         ctx: &'a dyn QueryContext,
     ) -> Result<BoxStream<'a, Result<Self::Output>>> {
+        let quota = ctx
+            .extensions()
+            .get::<QuotaTracking>()
+            .expect("`QuotaTracking` extension should be set during `ProContext` creation")
```

Comment:
Fingers crossed. I guess we can do this :thinking: … is this "the way" to use a type map in general?

Reply:
if you have any suggestions for how to pass the `Sender` to the quota channel let me know. This idea is inspired by the `Extensions` in actix.

Reply:
the thing is, it is not always there. It's only there if we are in Pro.

Reply:
Sure, but you are in a ProContext right? It could have a getter for that.

Reply:
We have the same `&dyn QueryContext` in free and in pro. We can't make that generic because we use the query processor as a trait object and trait objects can't have generic functions.

Reply:
You have to explain that to me why this is not possible. We also have other methods only in the ProContext, right?

Reply:
If the extensions are optional I would need to unwrap two times. One for the extensions and one for the thing I actually need.

What do you mean by parametrizing the QueryContext?

Reply:
I thought you would just return an error here instead of panicking and you are done.

Reply:
it's an error that can never happen, so why return it?

Reply:
Does having a QueryContext as a generic associated type help here?